File: 2ºCARRERA/tprog/practica2_872815_873338/c++/mul.cc
/*
 *  Asignatura : Tecnología de la Programación
 *  Práctica 2:
 *  Autores : Mateo Lorente, Diego  --- 873338
 *            Solana Melero, Carlos --- 872815
 *  Fichero : mul.cc
 */

#include "mul.h"

void Mul::ejecuta(stack<int32_t> &pila, int& pc) {
    int32_t num1,suma;
    suma = pila.top();
    pila.pop();
    suma *= pila.top();
    pila.pop();
    pila.push(suma);
    pc++;
}

void Mul::muestra(){
    cout << "mul";
}

File: 2ºCARRERA/tprog/practica2_872815_873338/java/Mul.java
/*
 *  Asignatura : Tecnología de la Programación
 *  Práctica 2:
 *  Autores : Mateo Lorente, Diego  --- 873338
 *            Solana Melero, Carlos --- 872815
 *  Fichero : Mul.java
 */

import java.util.Stack;

public class Mul implements Instruccion {
    public int ejecuta(Stack<Integer> pila, int pc) {
        int num1,suma;
        suma = pila.pop();
        suma = suma * pila.pop();
        pila.push(suma);
        pc++;
        return pc;
    }
    public void muestra() {
    System.out.print("mul");
    }
};

